setup.py

```diff
@@ -98,7 +102,16 @@ def build_cmake(self, extension):
             '--parallel' + ('' if cpu_number is None else ' ' + cpu_number)
         ]
 
-        self.spawn(['cmake', '-S', str(project_dir), '-B', str(build_temp)] + cmake_args)
+        update_needed = True
+        if (build_temp / 'CMakeCache.txt').exists():
+            cmake_files = glob.glob(str(project_dir / 'cmake' / 'externals' / '*'))
+            cmake_files += [str(project_dir / 'CMakeLists.txt'), str(build_temp / 'CMakeCache.txt')]
+            latest_file = max(cmake_files, key=os.path.getctime)
+            if latest_file.endswith('CMakeCache.txt'):
+                update_needed = False
+        if update_needed:
+            self.spawn(['cmake', '-S', str(project_dir), '-B', str(build_temp)] + cmake_args)
```

Comment:
@shaahji ,  Can you check the line 110 again? it means if the CMakeCache.txt is newest file among all the cmake files, the cmake updating process will be skiped (=False). I didn't see any problem here.
Also, it also checks the files under cmake/externals folder where the other cmake file are.

"CMake is smart enough to not trigger a reconfigure if not necessary. This change is hiding some deeper problem with the build setup"
is it true in your environment now?
At least the cmake updating process will take some time.

Reply:
> Can you check the line 110 again?

My bad, I misread the code.

> Also, it also checks the files under cmake/externals folder where the other cmake file are.

But there are other transitive dependencies that this check doesn't cover. For example, I could locally modify a OpenCV *.cmake file which this change would not pick up. From a user's perspective the build would succeed but the change would not have been picked up - false positives. Build dependencies are complicated problems and trying to solve by a few lines of python script is underestimating the complexity of the problem.

> is it true in your environment now?

You are taking my comment out of context. My comment also read "... this change is hiding a deeper problem with the build setup." CMake configure should not trigger a full project rebuild. I can run CMake configure and build operations interleaved any number of times and except for the first time, CMake will not trigger a project build unless otherwise there's a code change or the environment changes somehow. That is true if the build dependencies are setup correct. Build issues, more often than not, are a result of wrong dependency setup.

And, yes, I fixed the issue locally on my machine. It had to do with my local setting in CMakeSettings.json file. My intention was to append the git path to the environment variable path (required for FetchContent to work) which I mistyped it to `<path to git>;%PATH%` instead of `<path to git>;${env.PATH}` because CMakeSettings.json understands only its own macros. There was another issue with setup of Ninja as well - the order in which vcvars were setup and Ninja needed to be found was reversed.

Regardless, use of python as a _build tool_ has introduced too many variables (most undocumented) in the build process. With this change you are trying to extend that to deal with dependencies as well. I wouldn't recommend extending this any further than what it already is. That's my penny in the dollar!